src/Swift.Runtime/src/Swift/Runtime/InteropServices/SwiftMarshal.cs

```diff
@@ -15,22 +17,33 @@ public static class SwiftMarshal
     /// </summary>
     /// <typeparam name="T">The type of the value being marshaled</typeparam>
     /// <param name="value">The value to marshal</param>
-    /// <param name="swiftDest">the destination for marshaling</param>
+    /// <param name="swiftDestSpan">the destination for marshaling</param>
     /// <returns>A pointer to memory to pass in to Swift for marshaling. Note: this value may be different from the value passed in.</returns>
     /// <exception cref="NotSupportedException"></exception>
-    public static IntPtr MarshalToSwift<T>(T value, IntPtr swiftDest)
+    public static void MarshalToSwift<T>(T value, Span<byte> swiftDestSpan)
     {
         if (value is ISwiftObject swiftValue)
         {
-            return swiftValue.MarshalToSwift(new IntPtr(swiftDest));
+            swiftValue.MarshalToSwift(swiftDestSpan);
+            return;
         }
 
         var type = typeof(T);
         if (type.IsPrimitive || typeof(nint).IsAssignableFrom(type) || typeof(nuint).IsAssignableFrom(type))
         {
             unsafe
             {
-                return new IntPtr(MarshalPrimitiveToSwift(value, (void*)swiftDest));
+                var typeSize = Marshal.SizeOf<T>();
```

Comment:
I would assume that `T` is at least blittable? If not, we should discuss how this API is being used.

Reply:
Yes, T is blittable (except bool).

Reply:
What about `char` or enum values? Neither are blittable. I would also consider applying the `unmanaged` constraint to `T` which is what I think you're stating here.

If `T` is expected to be `unmanaged`, then you should just use `sizeof` and not engage the marshalling logic.

Reply:
Char and enum values shouldn't be marshalled as primitives but rather `ISwiftObject`. I replaced it with `Unsafe.SizeOf<T>();` as I can't apply `unmanaged` since T can be marshalled as `ISwiftObject`.

Reply:
Okay. Just be aware that `typeof(char).IsPrimitive` evaluates to `true` so it might be prudent to assert `T` isn't `char`.

Reply:
Added a condition to skip if `char`.